[ADD] API 호출하여 더미데이터 저장기능 구현

twkim5235 opened the pull request:
#2 
- 애플리케이션 시작시, 더미데이터 저장
- Product Entity의 멤버 변화로 연관된 모든 Class 변경
  - Category Entity 변경
  - Store 통해 저장시, Product 생성을 위한 Request 변경
- docker-compose 맥용 새로 작성

twkim5235 committed:
[ADD] API 호출하여 더미데이터 저장기능 구현
- 애플리케이션 시작시, 더미데이터 저장
- Product Entity의 멤버 변화로 연관된 모든 Class 변경
 - Category Entity 변경
 - Store 통해 저장시, Product 생성을 위한 Request 변경
- docker-compose 맥용 새로 작성

gratis-caelum reviewed (changes requested):
## 요약
- API 호출로 더미데이터 저장 기능 확인
- macOS 전용 docker-compose file 추가 확인

## 주요 요청 사항
1. **비동기 transaction 관리**
- `@Async` method 에 `@Transactional` or `CompletableFuture` 예외 처리 덧붙이면 좋을 것 같습니다.
2. ** 환경 변수 외부화 **
- `docker-compose-mac.yml` 에 DB Password `.env` 분리하는 게 좋아보입니다.

twkim5235 committed:
[FIX] DDL-AUTO 삭제
- 애플리케이션 실행할 때마다 스키자 생성부분 삭제

twkim5235 committed:
[FIX] 병찬님 코드리뷰 반영
- mysql password .env로 변경
- FetchProducts() @Transacional 어노테이션 추가
  - exception 발생 시, 트래잭션 rollback
- 프로그램을 시작할 때 마다, 데이터를 모두 저장하면 중복 발생으로 인해, 중복 방지 로직 추가

twkim5235 committed:
[FIX] Refactoring
- 테스트 환경 개선을 위해, ddl-auto update로 설정
- 이미지의 url의 길이가 255보다 길어 TEXT로 자료형 변경
